### PR TITLE
Routing

### PR DESCRIPTION
**Pull request by julibel**
*Summary*
Dashboard and Book Detail using Route React to redirect. 

https://trello.com/c/4cjDxkDa/6-routing

### CODE REVIEW

File: src/components/BookImg/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React, { Component } from 'react';
+
+export default class BookImg extends Component {
+  render() {
+    if (this.props.src === null){
```
Comment (pabloferro) [inline src/components/BookImg/index.js]: `if (!this.props.src) {` works too if src is [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)

File: src/screens/BookDetail/style.scss [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+.row {
+  display: flex;
+  flex-wrap: wrap;
+  margin-top: 16px;
+  max-width: 800px;
+  padding-left: 40px; 
+}
+
+.comments {
```
Comment (pabloferro) [inline src/screens/BookDetail/style.scss:119]: This should be in `Comment/style.scss`

File: src/screens/BookDetail/components/FutureComment/index.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React, { Component } from 'react';
+import './style.css';
+import profile from '../../../../assets/search.svg';
+import '../../../../style.css';
+
+export default class Comment extends Component {
```
Comment (pabloferro) [inline src/screens/BookDetail/components/FutureComment/index.js]: Fix class name.
Export default at the end of the file.

File: src/screens/BookDetail/index.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+          <FutureComment />
+          <Comment />
+          <Comment />
+        </div>
+      </div>
+    )
+  }
+  
+  render() {
+    let bookSelected =  books.filter (book => book.id === parseInt(this.props.match.params.id, 10));
```
Comment (pabloferro) [inline src/screens/BookDetail/index.js]: You can use `.find` and check if its defined.

```js
if (bookSelected) {
   ...
}
```

File: src/screens/BookDetail/index.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    if (bookSelected.length === 0) {
+      return <div>Sorry, but the book was not found</div>
+    }
+
+    return (
+      <div>
+        <Link className="volver" to='/'>&lt; Volver</Link>
+        {this.bookInfo(bookSelected[0])} 
+        {this.sugestions()}
+        {this.comments()}
```
Comment (pabloferro) [inline src/screens/BookDetail/index.js]: I'd move these three functions into the render.
I think they only make this component harder to understand.

File: src/screens/BookDetail/style.scss [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+.profile-img {
+  background: $dusty-gray;
+  border-radius: 100%; 
+  height: 40px;
+  margin-right: 16px;
+  width: 40px;
+}
+
+.volver {
```
Comment (pabloferro) [inline src/screens/BookDetail/style.scss]: Name classes in english

File: src/screens/Dashboard/components/BookList/index.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import React, { Component } from 'react';
+import BookImg from '../../../../components/BookImg';
+import { Link } from 'react-router-dom'; 
+import './style.css';
+
+export default class BooksList extends Component {
+  renderBook(book) {
+    return (
+      <Link className="book-link" to={`/books/${book.id}`}>
+        <div key={book.id}>
```
Comment (pabloferro) [inline src/screens/Dashboard/components/BookList/index.js]: key should be in the `Link` component

File: src/screens/Dashboard/components/BookList/index.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+          <BookImg src={book.image_url} classNameImg={"book-logo"} classNameNoImg={"no-img"} />
```
Comment (pabloferro) [inline src/screens/Dashboard/components/BookList/index.js]:
```js
classNameImg="book-logo" classNameNoImg="no-img"
```

File: src/screens/Dashboard/components/BookList/index.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+          <h2 className="book-title">{book.title}</h2>
+          <p className="book-author">{book.author}</p>
+        </div>
+      </Link>
+    )
+  }
+  
+  render() {
+    const books = this.props.books;
```
Comment (pabloferro) [inline src/screens/Dashboard/components/BookList/index.js]:
```js
const { books } = this.props;
```

But I wouldn't use a const here just for books.

**Review by pabloferro — CHANGES_REQUESTED** [review]

File: src/screens/BookDetail/style.scss (hunk 2 above)
Comment (julibel) [reply src/screens/BookDetail/style.scss:119]: Class .comments is used by Comment and FutureComment so it should be in that slyle.scss

File: src/screens/BookDetail/index.js [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+import books from '../../constants/books';
+import './style.css';
+import BookImg from '../../components/BookImg';
+import Comment from './components/Comment';
+import FutureComment from './components/FutureComment';
+import { Link } from 'react-router-dom'; 
+
+class BookDetail extends Component {
+  render() {
+    let bookSelected =  books.find (book => book.id === parseInt(this.props.match.params.id, 10));
```
Comment (pabloferro) [inline src/screens/BookDetail/index.js]: Use const if you don't ressign

**Review by pabloferro — APPROVED** [review]

**Pull request merged**